chain/chain/src/chain.rs

```diff
@@ -416,12 +416,26 @@ impl Chain {
     ///  * `header` - the last finalized block seen from `header` (not pushed back) will be used to
     ///               compute the light client block
     pub fn create_light_client_block(
-        header: &BlockHeader,
+        header: BlockHeader,
         runtime_adapter: &dyn RuntimeAdapter,
         chain_store: &mut dyn ChainStoreAccess,
     ) -> Result<LightClientBlockView, Error> {
-        let final_block_header =
-            chain_store.get_block_header(&header.inner_rest.last_final_block)?.clone();
+        let final_block_header = {
```

Comment:
Suggestions how to make it cleaner are very welcome :)

Reply:
When `two_ahead.inner_lite.epoch_id != ret.inner_lite.epoch_id`, why couldn't you set `new_final_hash` directly to `ret.inner_rest.last_final_block`?

Reply:
A1 <- A2 <- A3 <- B1 <- B2 <- B3 <- C

Letters represent epochs.

Say `C` is passed in. `ret` is `B2`.  `two_ahead` is `C`, it's in a different epoch. If I take the last final block as of `B2`, I get `A3`, which doesn't have two blocks on top of it in the same epoch.